tests/scripts/task_python_frontend.sh

```diff
@@ -31,6 +31,9 @@ find . -type f -path "*.pyc" | xargs rm -f
 # Rebuild cython
 make cython3
 
+# Only run GPU enabled tests on GPU
+export PYTEST_ADDOPTS="-m gpu $PYTEST_ADDOPTS"
```

Comment:
Are we sure we want this? I think this is skipping a lot of tests that were previously running. The GPU frontend test only took 1h 16 min which seems too fast to me https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/PR-7438/6/pipeline/390

It seems there are many tests that don't use `@tvm.testing.uses_gpu`. For example https://github.com/apache/tvm/blob/96b09817fc1796a789524ae30cd2d7e9d6f73d6c/tests/python/frontend/pytorch/test_object_detection.py#L94

Reply:
Probably we should use `targets` option in `verify_with_ort_with_inputs` for now https://github.com/apache/tvm/blob/3beec22264f56f734de7d14cd6382b96e83e280a/tests/python/frontend/onnx/test_forward.py#L126

See for example https://github.com/apache/tvm/blob/3beec22264f56f734de7d14cd6382b96e83e280a/tests/python/frontend/onnx/test_forward.py#L3421-L3423 (we should test this function on GPU, btw)

Reply:
Right now, all the tests are running on GPU, which causes the argmin / argmax onnx test to fail. The ONNX test file uses @tvm.testing.uses_gpu to indicate whether we should run tests on GPU or not, but right now it doesn't do anything, which is definitely not good. I think we should try to move towards enabling the @tvm.testing.uses_gpu in the long run.

What if I set PYTEST_ADDOPTS before the onnx tests, and reset it after? Since the file uses the @tvm.testing.uses_gpu a lot, I think we should either remove the decorators or let them actually do something.

Reply:
Overall, I think that hardcoding targets is bad because it's not easy to see -- that's what caused none of the ONNX tests to be run on GPU in the first place.

Reply:
ok, surprised to hear that `use_gpu` doesn't do anything. Let's make it

```
PYTEST_ADDOPTS="-m gpu $PYTEST_ADDOPTS" run_pytest cython python-frontend-onnx tests/python/frontend/onnx
```
This should only modify env vars when running the onnx test. 

Also, can you go through `TODO(mbrookhart)` and add `use_gpus`? There are about 10-15 of them.
https://github.com/apache/tvm/blob/3beec22264f56f734de7d14cd6382b96e83e280a/tests/python/frontend/onnx/test_forward.py#L277

Reply:
Doh! I thought I had removed all of those comments, but I must have only done it for the dynamic test files and missed the onnx file.

Reply:
I can open a separate PR to remove these.

Reply:
I already did it!